### PR TITLE
docs: grid pro tweaks and improvements PTL-1517

### PR DESCRIPTION
**Pull request by matteematt**
1. Align formatting with other docs pages
2. Add some more examples
3. Make live examples load automatically

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js [hunk 1]
```diff
@@ -53,14 +54,21 @@ function GridProExampleBase({
       gridOptions = modifyGridOptions(gridOptions);
 
       grid.current.gridOptions = gridOptions;
+
+			setIsLoaded(true);
```
Comment (MrBrunoWolff) [inline docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js:58]: nitpick: not sure about this formatting

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js [hunk 2]
```diff
@@ -53,14 +54,21 @@ function GridProExampleBase({
...
+    return () => clearTimeout(timer);
+  }, []);
+
   return (
     <CodeSection>
       <div style={{ flex: 1, display: 'flex', flexDirection: 'column', height: gridHeight }}>
-        <rapid-button onClick={loadGridOptions}>{buttonText}</rapid-button>
-        <rapid-grid-pro ref={grid}></rapid-grid-pro>
+				{!isLoaded ? (<LoadingRing />) : null}
+				<rapid-grid-pro ref={grid} ></rapid-grid-pro>
```
Comment (MrBrunoWolff) [inline docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js:71]: same here

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/index.mdx [hunk 3]
```diff
@@ -36,35 +37,6 @@ For more detailed information on API and configurations, please refer to the [AP
 - **State persistence**: ability to save and restore grid state.
 - **Row models**: support for both client-side and server-side [row models](https://www.ag-grid.com/javascript-data-grid/row-models/).
 
-## Getting Started
```
Comment (MrBrunoWolff) [inline docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/index.mdx:39]: question:

are we doing this in all other places?

Feels odd having this bits at the end, not everyone will start from seed/create, it's worth mentioning those but not sure about moving to the bottom of this page

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js (hunk 2 above)
Comment (matteematt) [reply docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/grid-pro_examples.js]: DOne

File: docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/index.mdx (hunk 3 above)
Comment (matteematt) [reply docs/001_develop/03_client-capabilities/005_grids/003_grid-pro/index.mdx:39]: Yeah we're trying to tell the story of each component at first and as most people will have it pre-installed we're leading with usage and examples. Plus once someone has installed it they don't need to see it as the first section every time they open the docs

Long term we can change it back through if we get feedback around it

**Review by MrBrunoWolff — APPROVED** [review]

**Pull request merged**